gix-fsck/tests/fsck/connectivity/mod.rs

```diff
@@ -0,0 +1,84 @@
+use std::ops::Deref;
+
+use gix_fsck::ConnectivityCheck;
+use gix_hash::ObjectId;
+use gix_hashtable::HashMap;
+use gix_object::Kind;
+use gix_testtools::once_cell::sync::Lazy;
+
+use crate::hex_to_id;
+
+fn check_missing<'a>(repo_name: &str, commits: impl IntoIterator<Item = &'a ObjectId>) -> HashMap<ObjectId, Kind> {
+    let db = {
+        let fixture_path = gix_testtools::scripted_fixture_read_only_standalone("make_test_repos.sh")
+            .expect("fixture path")
+            .join(repo_name)
+            .join(".git")
+            .join("objects");
+        let mut db = gix_odb::at(fixture_path).expect("odb handle");
+        db.refresh_never();
+        db
+    };
+
+    let mut missing: HashMap<ObjectId, Kind> = HashMap::default();
+    let callback = |oid: &ObjectId, kind: Kind| {
+        missing.try_insert(*oid, kind).expect("no duplicate oid");
+    };
+
+    let mut check = ConnectivityCheck::new(&db, callback);
+    for commit in commits.into_iter() {
+        check.check_commit(commit);
+    }
+
+    missing
+}
+
+fn hex_to_ids<'a>(hex_ids: impl IntoIterator<Item = &'a str>) -> Vec<ObjectId> {
+    hex_ids.into_iter().map(hex_to_id).collect()
+}
+
+fn hex_to_objects<'a>(hex_ids: impl IntoIterator<Item = &'a str>, kind: Kind) -> HashMap<ObjectId, Kind> {
```

Comment:
I think this would be called `hex_to_ids` to stay in line with `hex_to_id` which is used everywhere.
In theory, this utility could be provided by `gix-testtools` as well.

Reply:
This function doesn't just spit out a collection of `ObjectId`'s, but rather, it provides a map of `ObjectID`s to `Kind`'s. The function above this does exactly what you're implying, and it is named as you'd expect (`hex_to_ids`).

I understand that "objects" probably isn't the best naming here, as it is not outputting actual objects, but rather a representation of their (expected) metadata. Totally open to better naming!